ls1intum/Artemis
# `General`: Improve detailed feedback view for exercises

### Checklist
#### General


#### Server
#### Client

### Motivation and Context


The result detail view (when you click on 7 of 13 passed and see the test feedback) gets improved.

### Description

- The text about potentially hidden test feedback after the due date is now only shown before the latest individual due date
- If the static code analysis would deduct more points than is allowed by the settings in the gradings tab, the student correctly sees the deducted amount and an explanatory tooltip.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 2 Student
- 1 of each Exercise. (Programming exercise needs SCA feedback and Submissionpolicy and some manual feedback)

1. Log in to Artemis
2. Create the exercises
3. Participate as a student 
4. Grade the exerises
5. Check the feedback in the student view
6. Set the individual due date of the other student to some later date
7. Check that the student without the longer due date can see a notification about potentially hidden feedback at the top of the feedback view

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|
| result-detail.component.ts | 87.31% | 94.4% |
| ParticipationService.java | 78% | 94% |
| ProgrammingExerciseGradingService.java | 87% | 96% |

### Screenshots


![Bildschirmfoto 2022-06-08 um 21 22 39](https://user-images.githubusercontent.com/38322605/172699719-bad6e33a-e0d4-4777-a7f3-62b9088e8efb.png)
